kube/deployment_test.go

```diff
@@ -95,16 +95,101 @@ func TestNewDeploymentHelm(t *testing.T) {
 	assert.Equal(deployment.Get("metadata", "name").String(), "role")
 
 	t.Run("Defaults", func(t *testing.T) {
-		t.Parallel()
+		// t.Parallel()
 		// Rendering fails with defaults, template needs information
 		// about sizing and the like.
 		_, err = testhelpers.RenderNode(deployment, nil)
 		assert.EqualError(err,
 			`template: :9:17: executing "" at <fail "role must have...>: error calling fail: role must have at least 1 instances`)
 	})
 
+	t.Run("Configured, not enough replicas", func(t *testing.T) {
+		// t.Parallel()
+		config := map[string]interface{}{
+			"Values.sizing.role.count":                 "0",
+			"Values.sizing.role.affinity.nodeAffinity": "snafu",
+			"Values.sizing.role.capabilities":          []interface{}{},
+			"Values.kube.registry.hostname":            "docker.suse.fake",
+			"Values.kube.organization":                 "splat",
+			"Values.env.KUBE_SERVICE_DOMAIN_SUFFIX":    "domestic",
+		}
+		_, err = testhelpers.RenderNode(deployment, config)
```

Comment:
Your data race is because the same `err` is being assigned to

Reply:
Hm. Maybe. I will definitely fix that and hope for the best. However the stack traces go gave me show something deep in the template render code, not the assignment to err. Well, we will see.

Reply:
I believe the issue was that we were printing out giant YAML things when the templates failed to expand or something, on `stdout`, which was between lines of the stack on `stderr`, or something along those lines.